rust/query/typeql_match.rs

```diff
@@ -182,9 +182,10 @@ fn expect_variable_names_are_unique(conjunction: &Conjunction) -> Result<()> {
     let value_names = value_refs.iter().map(|r| r.name()).collect::<HashSet<_>>();
     let common_refs = concept_names.intersection(&value_names).collect::<HashSet<_>>();
     if !common_refs.is_empty() {
-        Err(TypeQLError::VariableNameConflict(common_refs.iter().map(|r| r.to_string()).join(", ")))?
+        Err(TypeQLError::VariableNameConflict(common_refs.iter().map(|r| r.to_string()).join(", ")).into())
+    } else {
+        Ok(())
     }
```

Comment:
Either remove the else and use explicit `return` for `Err`, or flip the condition. I'd go with the former, that sounds more like what we're trying to check for.